sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy

```diff
@@ -95,222 +93,345 @@ class APISpec extends Specification {
-    static File getRandomFile(int size) {
-        File file = File.createTempFile(UUID.randomUUID().toString(), ".txt")
-        file.deleteOnExit()
-        FileOutputStream fos = new FileOutputStream(file)
-        fos.write(getRandomData(size).array())
-        fos.close()
-        return file
+        if (testMode == TestMode.RECORD) {
+            builder.addPolicy(interceptorManager.getRecordPolicy())
+        }
+
+        builder.credential(credential).buildBlobAsyncClient()
     }
 
-    static long getRandomSeed() {
-        return System.currentTimeMillis()
+    BlobClient getBlobClient(SASTokenCredential credential, String endpoint, String blobName) {
+        return getBlobClient(credential, endpoint, blobName, null)
     }
 
-    def setupSpec() {
-        /*
-        We'll let primary creds throw and crash if there are no credential specified because everything else will fail.
-         */
-        primaryCreds = getGenericCreds("PRIMARY_STORAGE_")
-
-        /*
-        It's feasible someone wants to test a specific subset of tests, so we'll still attempt to create each of the
-        ServiceURLs separately. We don't really need to take any action here, as we've already reported to the user,
-        so we just swallow the exception and let the relevant tests fail later. Perhaps we can add annotations or
-        something in the future.
-         */
-        try {
-            alternateCreds = getGenericCreds("SECONDARY_STORAGE_")
-            alternateServiceURL = getGenericServiceURL(alternateCreds)
+    BlobClient getBlobClient(SASTokenCredential credential, String endpoint, String blobName, String snapshotId) {
+        BlobClientBuilder builder = new BlobClientBuilder()
+            .endpoint(endpoint)
+            .blobName(blobName)
+            .snapshot(snapshotId)
+            .httpClient(getHttpClient())
+            .httpLogDetailLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+
+        if (testMode == TestMode.RECORD) {
+            builder.addPolicy(interceptorManager.getRecordPolicy())
+        }
+
+        return builder.credential(credential).buildBlobClient()
+    }
+
+    BlobClient getBlobClient(SharedKeyCredential credential, String endpoint, HttpPipelinePolicy... policies) {
+        BlobClientBuilder builder = new BlobClientBuilder()
+            .endpoint(endpoint)
+            .httpClient(getHttpClient())
+            .httpLogDetailLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+
+        for (HttpPipelinePolicy policy : policies) {
+            builder.addPolicy(policy)
         }
-        catch (Exception e) {
+
+        if (testMode == TestMode.RECORD) {
+            builder.addPolicy(interceptorManager.getRecordPolicy())
         }
 
-        try {
-            blobStorageServiceURL = getGenericServiceURL(getGenericCreds("BLOB_STORAGE_"))
+        return builder.credential(credential).buildBlobClient()
+    }
+
+    BlobClient getBlobClient(SharedKeyCredential credential, String endpoint, String blobName) {
```

Comment:
Since you have several different clients depend on same builder. Could you make the helper method for the builder instead of client?

Reply:
Will look into this change after the PR goes through.